### PR TITLE
update lightning version

### PR DESCRIPTION
**Pull request by fabianp**

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -13,19 +13,19 @@ source:
   {{ hash_type }}: {{ hash_val }}
 
 build:
-  number: 1
+  number: 0
```
Comment (jakirkham) [inline recipe/meta.yaml:16]: Would bump the build number as well too as bad packages may have leaked out from the old PR ( https://github.com/conda-forge/sklearn-contrib-lightning-feedstock/pull/4 ).

Comment (fabianp) [reply recipe/meta.yaml:16]: what should I put here? I just want to update the version with the latest release ...

**Comment by aldanor** [comment]
@fabianp Any plans to merge this? 0.5 has been out for a while now...

**Comment by fabianp** [comment]
appveyor is failing, and I have no idea how to fix it...

**Comment by jakirkham** [comment]
@conda-forge-admin, please rerender.

**Comment by aldanor** [comment]
That’s one very weird error. It builds and installs fine and at the very end something distutils-related blows up.

**Comment by aldanor** [comment]
Any ideas? Never seen this kind of error before, and can’t seem to google anything related. Hm...

**Comment by aldanor** [comment]
Can we just disable the Windows build then if no one can fix it? Having OS X and Linux built and published is better than none at all.

**Comment by vene** [comment]
Seems like in the 0.5.0 sdist, which I made, the cpp files generated by cython contain absolute paths to numpy headers on my machine.

EDIT: actually the problem is SOURCES.txt, not the cpp files.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -13,19 +13,19 @@ source:
...
   script: python setup.py install --single-version-externally-managed --record record.txt
```
Comment (vene) [inline recipe/meta.yaml:17]: Quick and dirty fix: change this line to
```
  script: rmdir /S /Q sklearn_contrib_lightning.egg-info & python setup.py install --single-version-externally-managed --record record.txt
```

**Comment by vene** [comment]
Deleting the egg-info subdirectory inside the sdist tarball seems to correct this issue. I left an inline comment with a possible quick fix, although the right thing to be is probably to make a new release, making sure tarballs that don't include that directory. (I don't know how to do this.)

I don't know if there are any negative effects or not to doing this: the conda-build command runs for me and the tests almost work (except for l1l2 debiasing:)

```
{lamb} "C:\bld\sklearn-contrib-lightning_1522949533840\_test_env\python.exe" -s "C:\bld\sklearn-contrib-lightning_1522949533840\test_tmp\run_test.py"
===== testing package: sklearn-contrib-lightning-0.5.0-np111py35_2 =====
running run_test.py
.............................................................F.....................................................................
======================================================================
FAIL: lightning.impl.tests.test_primal_cd.test_debiasing_l1l2
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\bld\sklearn-contrib-lightning_1522949533840\_test_env\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "C:\bld\sklearn-contrib-lightning_1522949533840\_test_env\lib\site-packages\lightning\impl\tests\test_primal_cd.py", line 184, in test_debiasing_l1l2
    assert_equal(clf.n_nonzero(percentage=True), 0.08)
AssertionError: 0.070000000000000007 != 0.08

----------------------------------------------------------------------
Ran 131 tests in 5.781s

FAILED (failures=1)
===== sklearn-contrib-lightning-0.5.0-np111py35_2 OK =====
import: 'lightning'
```

**Comment by vene** [comment]
https://github.com/numpy/numpy/issues/7127 discussion here suggests that indeed sdists shouldn't contain egg-info.

**Comment by vene** [comment]
I believe the solution is to avoid importing setuptools in lightning if the command is sdist, as here: https://github.com/scikit-learn/scikit-learn/blob/master/setup.py#L48-L72

**Comment by jakirkham** [comment]
We [use `pip`]( https://conda-forge.org/docs/meta.html#use-pip ) for Python packages now. Maybe try that instead?

**Comment by vene** [comment]
By itself, switching to pip does not fix the issue: the presence of the incorrectly-constructed egg-info dir breaks windows builds whether with pip or with setup.py.

I believe this should fix future sdists: https://github.com/scikit-learn-contrib/lightning/pull/123

**Comment by jakirkham** [comment]
Pip doesn’t use egg-info. It uses dist-info. Maybe that runs into similar issues, but it’s worth checking (if you haven’t already).

**Pull request closed without merging**